lang/go/go.talon

```diff
@@ -66,25 +113,46 @@ else <user.text> [over]:
     key("enter")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
-state while: "while "
+[state] if (err|error):
+    insert("if err != nil {")
+    key("enter")
+
+[state] if not (err|error):
+    insert("if err == nil {")
+    key("enter")
+
 while <user.text> [over]:
-    insert("while ")
-    insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
+    user.insert_between("for "," {}")
 
 state for: "for "
-for <user.text> [over]:
+state for <user.text> [over]:
     insert("for ")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
+snip for range: "forr"
+
 state for range: "forr "
 range <user.text> [over]:
     insert("forr ")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
+[state] for i:
```

Comment:
I'd make the "state" mandatory here to not misfire.  I'd also be inclined to have this one be in a more standard location (eg in the imperative tag) and implemented via an action, as this is something that appears across many languages.